### PR TITLE
snapshot sysfs/procfs inside must-gather

### PR DESCRIPTION
**Pull request by fromanirh**
Signed-off-by: Francesco Romani <fromani@redhat.com>

### CODE REVIEW

**Comment by fromanirh** [comment]
/hold
placeholder/demo. Requires more GHW changes (new PR to be sent ASAP)

**Comment by fromanirh** [comment]
requires: https://github.com/openshift/release/pull/14777

File: go.sum [hunk 1]
```diff
@@ -231,6 +233,8 @@ github.com/fatih/color v1.7.0 h1:DkWD4oS2D8LGGgTQ6IvwJJXSL5Vp2ffcQg58nFV38Ys=
 github.com/fatih/color v1.7.0/go.mod h1:Zm6kSWBoL9eyXnKyktHP6abPY2pDugNf5KwzbycvMj4=
 github.com/flynn/go-shlex v0.0.0-20150515145356-3f9db97f8568/go.mod h1:xEzjJPgXI435gkrCt3MPfRiAkVrwSbHsst4LCFVfpJc=
 github.com/fogleman/gg v1.2.1-0.20190220221249-0403632d5b90/go.mod h1:R/bRT+9gY/C5z7JzPU0zXsXHKM4/ayA+zqcVNZzPa1k=
+github.com/fromanirh/ghw v0.0.15 h1:L0vAKHbrkpXr2Ap1nbhmagsw8aURX3DgZu1qMym4aIY=
+github.com/fromanirh/ghw v0.0.15/go.mod h1:QOXppNRCLGYR1H+hu09FxZPqjNt09bqUZUnOL3Rcero=
```
Comment (fromanirh) [inline go.sum:237]: this needs to go away

**Comment by fromanirh** [comment]
/hold
to resolve a catch-22 with CI, let's see if https://github.com/openshift-kni/performance-addon-operators/pull/505 helps

File: tools/gather-sysinfo/gather-sysinfo.go [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+	if *debug {
+		snapshot.SetTraceFunction(func(msg string, args ...interface{}) {
+			log.Printf(msg, args...)
+		})
+	}
+
+	scratchDir, err := ioutil.TempDir("", "perf-must-gather-*")
+	if err != nil {
+		log.Fatalf("error creating temporary directory: %v", err)
```
Comment (marcel-apf) [inline tools/gather-sysinfo/gather-sysinfo.go:29]: then we need to bail out?

File: tools/gather-sysinfo/gather-sysinfo.go [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+	}
+	defer os.RemoveAll(scratchDir)
+
+	if err := snapshot.CloneTreeInto(scratchDir); err != nil {
+		log.Fatalf("error cloning into %q: %v", scratchDir, err)
```
Comment (marcel-apf) [inline tools/gather-sysinfo/gather-sysinfo.go:34]: same here?

File: tools/gather-sysinfo/gather-sysinfo.go (hunk 2 above)
Comment (fromanirh) [reply tools/gather-sysinfo/gather-sysinfo.go:29]: yes we do: https://golang.org/src/log/log.go?s=7078:7134#L196

File: tools/gather-sysinfo/gather-sysinfo.go (hunk 3 above)
Comment (fromanirh) [reply tools/gather-sysinfo/gather-sysinfo.go:34]: ditto

**Comment by fromanirh** [comment]
/hold cancel
CI is happy, ready for final review

**Pull request merged**